src/overlays/actors/ovl_En_Blkobj/z_en_blkobj.c

```diff
@@ -27,23 +31,157 @@ const ActorInit En_Blkobj_InitVars = {
     (ActorFunc)EnBlkobj_Update,
     (ActorFunc)EnBlkobj_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Blkobj/func_809C2060.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Blkobj/EnBlkobj_Init.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F(scale, 1, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 800, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 200, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 300, ICHAIN_STOP),
+};
+
+UNK_TYPE D_809C2590[] = {
+    0xE200001C,
+    0xC8112078,
+    0xDF000000,
+    0x00000000,
+};
+
+UNK_TYPE D_809C25A0[] = {
+    0xE200001C,
+    0xC81049D8,
+    0xDF000000,
+    0x00000000,
+};
+
+extern UNK_PTR D_060014E0;
+extern UNK_PTR D_060053D0;
+extern UNK_PTR D_06007564;
+
+void func_809C2060(EnBlkobj* this, EnBlkobjActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+    this->unk_166 = 0;
+}
+
+void EnBlkobj_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnBlkobj* this = THIS;
+    s32 localC = 0;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyInfo_SetActorMove(&this->dyna, 0);
+    if (Flags_GetClear(globalCtx, this->dyna.actor.room)) {
+        this->unk_164 = 0xFF;
+        func_809C2060(this, func_809C22F4);
+    } else {
+        DynaPolyInfo_Alloc(&D_06007564, &localC);
+        this->dyna.dynaPolyId =
+            DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, localC);
+        func_809C2060(this, func_809C2148);
+    }
+}
+
+void EnBlkobj_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnBlkobj* this = THIS;
+
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+}
+
+void func_809C2148(EnBlkobj* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (this->dyna.actor.xzDistFromLink < 120.0f) {
+        func_809C2060(this, func_809C21A0);
+    }
+    player->stateFlags2 |= 0x4000000;
+}
+
+void func_809C21A0(EnBlkobj* this, GlobalContext* globalCtx) {
+    if (!(this->dyna.actor.flags & 0x40)) {
+        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_TORCH2, this->dyna.actor.posRot.pos.x,
+                    this->dyna.actor.posRot.pos.y, this->dyna.actor.posRot.pos.z, 0, this->dyna.actor.yawTowardsLink, 0,
+                    0);
+        func_809C2060(this, func_809C2218);
+    }
+}
+
+void func_809C2218(EnBlkobj* this, GlobalContext* globalCtx) {
+    s32 temp;
+
+    if (this->unk_166 == 0) {
+        if (Actor_Find(&globalCtx->actorCtx, ACTOR_EN_TORCH2, ACTORTYPE_BOSS) == NULL) {
+            Flags_SetClear(globalCtx, this->dyna.actor.room);
+            this->unk_166 += 1;
+        }
+    } else {
+        if ((this->unk_166++ < 0x65) ^ 1) {
```

Comment:
this line is suspect..does this make sense?

Reply:
You can probably try things like `if (!(this->unk_166++ < 0x65))`

Reply:
hmm, those don't work :/

Reply:
Yeah I don't know if those would work exactly, but that's the kind of stuff you want to try afaik.